File: engine/model/src/main/scala/tools/packet/guild/ShowGuildRanks.scala
package tools.packet.guild

import client.database.data.GuildData
import net.opcodes.SendOpcode
import tools.packet.PacketInput

class ShowGuildRanks(private var _npcId: Int, private var _ranks: java.util.List[GuildData]) extends PacketInput {
  def npcId: Int = _npcId

  def ranks: java.util.List[GuildData] = _ranks

  override def opcode(): SendOpcode = SendOpcode.GUILD_OPERATION
}

File: engine/engine-model/src/main/java/tools/packet/guild/ShowGuildRanks.java
package tools.packet.guild;

import java.util.List;

import client.database.data.GuildRankData;
import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record ShowGuildRanks(Integer npcId, List<GuildRankData> ranks) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.GUILD_OPERATION;
   }
}